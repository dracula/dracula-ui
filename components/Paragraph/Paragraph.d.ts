import React from 'react';
import { TextProps } from '../Text/Text';
/**
 * Paragraph is a semantic component used for blocks of text with
 * semantic meaning.
 *
 * Paragraph accepts all the same customization options as Text.
 */
export declare const Paragraph: React.FC<TextProps>;
