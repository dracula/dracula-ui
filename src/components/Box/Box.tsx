import cx from 'classnames/dedupe'
import React, { AllHTMLAttributes } from 'react'
import { borderColors, colors, glowColors } from '../../base/colors'
import {
  PaddingMixin,
  paddingMixin,
  marginMixin,
  MarginMixin
} from '../../base/spacing'

type Element = HTMLElementTagNameMap

export const roundedBorders = {
  none: 'drac-rounded-none',
  sm: 'drac-rounded-sm',
  base: 'drac-rounded',
  lg: 'drac-rounded-lg',
  xl: 'drac-rounded-xl',
  '2xl': 'drac-rounded-2xl',
  '3xl': 'drac-rounded-3xl',
  full: 'drac-rounded-full'
}

/**
 * Box Props
 */
export type BoxProps<K extends keyof Element = 'div'> = {
  /** The background color. */
  color?: keyof typeof colors

  /** The glow color. */
  glowColor?: keyof typeof glowColors

  /** The border color. */
  borderColor?: keyof typeof borderColors

  /** The border radius. */
  rounded?: keyof typeof roundedBorders

  /** The HTML element to be used */
  as?: K
} & AllHTMLAttributes<K> &
  PaddingMixin &
  MarginMixin

/**
 * Box is the most primitive component of Dracula UI.
 * Using Box allows for consumers of the components library to compose
 * more complex patterns, components, and UIs.
 *
 * Box includes built-in Color and Spacing properties that make building
 * complex components convenient and consistent.
 */
export function Box<T extends keyof Element>(props: BoxProps<T>) {
  const finalProps = {
    ...props,
    className: cx(
      `drac-box`,
      props.className,
      props.color && colors[props.color],
      props.glowColor && glowColors[props.glowColor],
      props.borderColor && borderColors[props.borderColor],
      props.rounded && roundedBorders[props.rounded],
      ...paddingMixin(props),
      ...marginMixin(props)
    )
  }

  const as = finalProps.as ?? 'div'
  return React.createElement(as, finalProps, props.children)
}

Box.displayName = 'Box'
