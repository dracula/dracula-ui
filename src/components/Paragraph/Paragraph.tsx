import cx from 'classnames/dedupe'
import React from 'react'
import { cleanProps, marginMixin, paddingMixin } from '../../base/spacing'
import {
  textColors,
  TextProps,
  textSizes,
  lineHeights,
  textWeights
} from '../Text/Text'

/**
 * Paragraph is a semantic component used for blocks of text with
 * semantic meaning.
 *
 * Paragraph accepts all the same customization options as Text.
 */
export const Paragraph: React.FC<TextProps> = (props: TextProps) => {
  const { size, weight, lineHeight, color, ...htmlProps } = props

  const finalProps = {
    ...htmlProps,

    className: cx(
      `drac-text`,
      props.className,
      textSizes[size ?? 'medium'],
      textWeights[weight ?? 'normal'],
      lineHeights[lineHeight ?? 'normal'],
      textColors[color ?? 'white'],
      ...paddingMixin(props),
      ...marginMixin(props)
    )
  }

  return <p {...cleanProps(finalProps)}>{props.children}</p>
}
