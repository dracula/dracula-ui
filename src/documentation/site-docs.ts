import {
  ComponentExample,
  renderSnapshot,
  SnapshotBuilder
} from './render-component'
import React from 'react'
import fs from 'fs-extra'
import { pretty } from './pretty'
import { getDocGen } from '../documentation/docgen/doc-generator'
import { toDSP } from '../documentation/dsp/component-generator'
import puppeteer from 'puppeteer'
import { componentScreenshot } from './screenshots/component-screenshot'
import os from 'os'

interface Documentation {
  basic: () => SnapshotBuilder
  variations: () => Array<SnapshotBuilder> | SnapshotBuilder
}

jest.setTimeout(10000)

const RECORDING = process.env.RECORDING

export function docs<T>(
  componentClass: React.FunctionComponent<T> | string,
  documentation: Documentation
) {
  const name =
    typeof componentClass === 'string'
      ? componentClass
      : componentClass.displayName ?? componentClass.name

  const dspPath = `${process.cwd()}/dsp`

  describe(`Site: ${name}`, () => {
    let browser: puppeteer.Browser

    beforeAll(async () => {
      if (!RECORDING) {
        return
      }

      let executablePath = undefined
      if (os.arch() === 'arm64') {
        executablePath = '/Applications/Chromium.app/Contents/MacOS/Chromium'
      }

      browser = await puppeteer.launch({
        headless: true,
        devtools: true,
        executablePath
      })
    })

    let examples: Record<string, ComponentExample> = {}

    afterAll(async () => {
      if (!RECORDING) {
        return
      }

      await browser.close()
      const docGen = getDocGen(name)
      const dsp = toDSP(name, examples, docGen)

      await fs.mkdirp(`${dspPath}/data/components/`).catch(() => {
        // folder already exists
      })

      const path = `${dspPath}/data/components/${name}.json`
      await fs.writeFile(path, pretty(JSON.stringify(dsp), 'json'))
    })

    test('Basic Usage', async () => {
      const variation = documentation.basic()
      const snapshot = renderSnapshot(
        variation.title,
        variation.comp,
        variation.docs
      )
      expect(snapshot).toMatchSnapshot()

      if (RECORDING) {
        const [screenshot, svg] = await componentScreenshot(
          browser,
          snapshot,
          null,
          name
        )

        snapshot.screenshot = screenshot
        snapshot.svgData = svg
      }

      examples = {
        ...examples,
        basic: snapshot
      }
    })

    const variations: SnapshotBuilder[] = ([] as SnapshotBuilder[]).concat(
      documentation.variations()
    )

    variations.forEach((variation) => {
      test(variation.title, async () => {
        const snapshot = renderSnapshot(
          variation.title,
          variation.comp,
          variation.docs
        )
        expect(snapshot).toMatchSnapshot()

        if (RECORDING) {
          const [screenshot, svg] = await componentScreenshot(
            browser,
            snapshot,
            variation,
            name
          )

          snapshot.screenshot = screenshot
          snapshot.svgData = svg
        }

        examples = {
          ...examples,
          [variation.title]: snapshot
        }
      })
    })
  })
}
