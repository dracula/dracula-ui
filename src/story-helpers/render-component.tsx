import React from 'react'
import Enzyme, { mount, shallow } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import { pretty } from './pretty'

Enzyme.configure({ adapter: new Adapter() })

export type ComponentExample = {
  html: string
  docs: string
  title: string
  react?: any
  screenshot?: string
  svgData?: string
}

export type SnapshotBuilder = {
  comp: () => React.ReactElement
  docs: string
  title: string
}

export function buildSnapshot(
  title: string,
  comp: () => React.ReactElement,
  docs: string = ''
): SnapshotBuilder {
  return {
    title,
    comp,
    docs: docs.trim()
  }
}

export function renderSnapshot(
  title: string,
  comp: () => React.ReactElement,
  docs: string = ''
): ComponentExample {
  let component = comp()
  const html = mount(component).html()
  let wrapped = false

  // wrap the component in a div when it's not wrapped already
  // we do this so we can get the actual component example to not be shallow rendered.
  // Shallow rendering a parent "div" gives us a nice JSX representation of the example.
  if (component.type !== 'div') {
    component = <div>{component}</div>
    wrapped = true
  }

  let react = shallow(component).debug({ verbose: true })

  if (wrapped) {
    // replace wrapper div ðŸ˜¬
    react = react
      .replace(/^<div>\n/, '')
      .replace(/\n<\/div>$/, '')
      .trim()
  }

  return {
    title,
    html: pretty(html, 'html'),
    react: pretty(react, 'typescript'),
    docs
  }
}
