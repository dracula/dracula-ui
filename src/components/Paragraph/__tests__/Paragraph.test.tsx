import React from 'react'
import { Paragraph } from '@/components/Paragraph/Paragraph'
import { docs } from '@/documentation/site-docs'
import { snapshot } from '../../../documentation/render-component'
import { Box } from '../../Box/Box'

docs(Paragraph, {
  basic() {
    return snapshot(
      'Usage',
      () => <Paragraph>basic</Paragraph>,
      'Show a paragraph within an interface.'
    )
  },
  variations() {
    return [
      snapshot(
        'Colors',
        () => (
          <Box>
            <Paragraph color="cyan">cyan</Paragraph>
            <Paragraph color="green">green</Paragraph>
          </Box>
        ),
        'Paragraphs can be customized to use any of the Dracula UI theme colors.'
      ),
      snapshot(
        'Weights',
        () => (
          <Box>
            <Paragraph weight="normal">normal</Paragraph>
            <Paragraph weight="semibold">semibold</Paragraph>
            <Paragraph weight="bold">bold</Paragraph>
          </Box>
        ),
        'Paragraphs can be defined in different font weights.'
      ),
      snapshot(
        'Sizes',
        () => (
          <Box>
            <Paragraph size="xs">xs</Paragraph>
            <Paragraph size="sm">sm</Paragraph>
            <Paragraph size="md">md</Paragraph>
            <Paragraph size="lg">lg</Paragraph>
          </Box>
        ),
        'Paragraphs can be customize to use several different sizes.'
      ),
      snapshot(
        'Line Height',
        () => (
          <Box>
            <Paragraph color="red">xs</Paragraph>
            <Paragraph lineHeight="xs">Zombie ipsum reversus ab viral inferno, nam rick grimes malum cerebro. De carne lumbering animata corpora quaeritis. Summus brains sit​​, morbo vel maleficia? De apocalypsi gorger omero undead survivor dictum mauris. Hi mindless mortuis soulless creaturas, imo evil stalking monstra adventus resi dentevil vultus comedat cerebella viventium. Qui animated corpse, cricket bat max brucks terribilem incessu zomby. The voodoo sacerdos flesh eater, suscitat mortuos comedere carnem virus. Zonbi tattered for solum oculi eorum defunctis go lum cerebro.</Paragraph>

            <Paragraph color="orange">sm</Paragraph>
            <Paragraph lineHeight="sm">Zombie ipsum reversus ab viral inferno, nam rick grimes malum cerebro. De carne lumbering animata corpora quaeritis. Summus brains sit​​, morbo vel maleficia? De apocalypsi gorger omero undead survivor dictum mauris. Hi mindless mortuis soulless creaturas, imo evil stalking monstra adventus resi dentevil vultus comedat cerebella viventium. Qui animated corpse, cricket bat max brucks terribilem incessu zomby. The voodoo sacerdos flesh eater, suscitat mortuos comedere carnem virus. Zonbi tattered for solum oculi eorum defunctis go lum cerebro.</Paragraph>

            <Paragraph color="yellow">md</Paragraph>
            <Paragraph lineHeight="md">Zombie ipsum reversus ab viral inferno, nam rick grimes malum cerebro. De carne lumbering animata corpora quaeritis. Summus brains sit​​, morbo vel maleficia? De apocalypsi gorger omero undead survivor dictum mauris. Hi mindless mortuis soulless creaturas, imo evil stalking monstra adventus resi dentevil vultus comedat cerebella viventium. Qui animated corpse, cricket bat max brucks terribilem incessu zomby. The voodoo sacerdos flesh eater, suscitat mortuos comedere carnem virus. Zonbi tattered for solum oculi eorum defunctis go lum cerebro.</Paragraph>

            <Paragraph color="pink">lg</Paragraph>
            <Paragraph lineHeight="lg">Zombie ipsum reversus ab viral inferno, nam rick grimes malum cerebro. De carne lumbering animata corpora quaeritis. Summus brains sit​​, morbo vel maleficia? De apocalypsi gorger omero undead survivor dictum mauris. Hi mindless mortuis soulless creaturas, imo evil stalking monstra adventus resi dentevil vultus comedat cerebella viventium. Qui animated corpse, cricket bat max brucks terribilem incessu zomby. The voodoo sacerdos flesh eater, suscitat mortuos comedere carnem virus. Zonbi tattered for solum oculi eorum defunctis go lum cerebro.</Paragraph>

            <Paragraph color="cyan">xl</Paragraph>
            <Paragraph lineHeight="xl">Zombie ipsum reversus ab viral inferno, nam rick grimes malum cerebro. De carne lumbering animata corpora quaeritis. Summus brains sit​​, morbo vel maleficia? De apocalypsi gorger omero undead survivor dictum mauris. Hi mindless mortuis soulless creaturas, imo evil stalking monstra adventus resi dentevil vultus comedat cerebella viventium. Qui animated corpse, cricket bat max brucks terribilem incessu zomby. The voodoo sacerdos flesh eater, suscitat mortuos comedere carnem virus. Zonbi tattered for solum oculi eorum defunctis go lum cerebro.</Paragraph>
          </Box>
        ),
        'Paragraphs can be controlled with different line heights.'
      )
    ]
  }
})
