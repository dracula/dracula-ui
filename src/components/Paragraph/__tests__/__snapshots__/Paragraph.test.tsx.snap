// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site: Paragraph Basic Usage 1`] = `
Object {
  "docs": "Show a paragraph within an interface.",
  "html": "<p class=\\"drac-text drac-line-height drac-text-white\\">basic</p>
",
  "react": "<Paragraph>basic</Paragraph>;
",
  "title": "Usage",
}
`;

exports[`Site: Paragraph Colors 1`] = `
Object {
  "docs": "Paragraphs can be customized to use any of the Dracula UI theme colors.",
  "html": "<div class=\\"drac-box\\">
  <p class=\\"drac-text drac-line-height drac-text-cyan\\">cyan</p>
  <p class=\\"drac-text drac-line-height drac-text-green\\">green</p>
</div>
",
  "react": "<Box>
  <Paragraph color=\\"cyan\\">cyan</Paragraph>
  <Paragraph color=\\"green\\">green</Paragraph>
</Box>;
",
  "title": "Colors",
}
`;

exports[`Site: Paragraph Line Height 1`] = `
Object {
  "docs": "Paragraphs can be controlled with different line heights.",
  "html": "<div class=\\"drac-box\\">
  <p class=\\"drac-text drac-line-height drac-text-red\\">none</p>
  <p class=\\"drac-text drac-line-height-none drac-text-white\\">
    Zombie ipsum reversus ab viral inferno, nam rick grimes malum cerebro. De
    carne lumbering animata corpora quaeritis. Summus brains sit​​, morbo vel
    maleficia? De apocalypsi gorger omero undead survivor dictum mauris. Hi
    mindless mortuis soulless creaturas, imo evil stalking monstra adventus resi
    dentevil vultus comedat cerebella viventium. Qui animated corpse, cricket
    bat max brucks terribilem incessu zomby. The voodoo sacerdos flesh eater,
    suscitat mortuos comedere carnem virus. Zonbi tattered for solum oculi eorum
    defunctis go lum cerebro.
  </p>
  <p class=\\"drac-text drac-line-height drac-text-orange\\">small</p>
  <p class=\\"drac-text drac-line-height-sm drac-text-white\\">
    Zombie ipsum reversus ab viral inferno, nam rick grimes malum cerebro. De
    carne lumbering animata corpora quaeritis. Summus brains sit​​, morbo vel
    maleficia? De apocalypsi gorger omero undead survivor dictum mauris. Hi
    mindless mortuis soulless creaturas, imo evil stalking monstra adventus resi
    dentevil vultus comedat cerebella viventium. Qui animated corpse, cricket
    bat max brucks terribilem incessu zomby. The voodoo sacerdos flesh eater,
    suscitat mortuos comedere carnem virus. Zonbi tattered for solum oculi eorum
    defunctis go lum cerebro.
  </p>
  <p class=\\"drac-text drac-line-height drac-text-yellow\\">normal</p>
  <p class=\\"drac-text drac-line-height drac-text-white\\">
    Zombie ipsum reversus ab viral inferno, nam rick grimes malum cerebro. De
    carne lumbering animata corpora quaeritis. Summus brains sit​​, morbo vel
    maleficia? De apocalypsi gorger omero undead survivor dictum mauris. Hi
    mindless mortuis soulless creaturas, imo evil stalking monstra adventus resi
    dentevil vultus comedat cerebella viventium. Qui animated corpse, cricket
    bat max brucks terribilem incessu zomby. The voodoo sacerdos flesh eater,
    suscitat mortuos comedere carnem virus. Zonbi tattered for solum oculi eorum
    defunctis go lum cerebro.
  </p>
  <p class=\\"drac-text drac-line-height drac-text-pink\\">medium</p>
  <p class=\\"drac-text drac-line-height-md drac-text-white\\">
    Zombie ipsum reversus ab viral inferno, nam rick grimes malum cerebro. De
    carne lumbering animata corpora quaeritis. Summus brains sit​​, morbo vel
    maleficia? De apocalypsi gorger omero undead survivor dictum mauris. Hi
    mindless mortuis soulless creaturas, imo evil stalking monstra adventus resi
    dentevil vultus comedat cerebella viventium. Qui animated corpse, cricket
    bat max brucks terribilem incessu zomby. The voodoo sacerdos flesh eater,
    suscitat mortuos comedere carnem virus. Zonbi tattered for solum oculi eorum
    defunctis go lum cerebro.
  </p>
  <p class=\\"drac-text drac-line-height drac-text-cyan\\">large</p>
  <p class=\\"drac-text drac-line-height-lg drac-text-white\\">
    Zombie ipsum reversus ab viral inferno, nam rick grimes malum cerebro. De
    carne lumbering animata corpora quaeritis. Summus brains sit​​, morbo vel
    maleficia? De apocalypsi gorger omero undead survivor dictum mauris. Hi
    mindless mortuis soulless creaturas, imo evil stalking monstra adventus resi
    dentevil vultus comedat cerebella viventium. Qui animated corpse, cricket
    bat max brucks terribilem incessu zomby. The voodoo sacerdos flesh eater,
    suscitat mortuos comedere carnem virus. Zonbi tattered for solum oculi eorum
    defunctis go lum cerebro.
  </p>
  <p class=\\"drac-text drac-line-height drac-text-green\\">xlarge</p>
  <p class=\\"drac-text drac-line-height-xl drac-text-white\\">
    Zombie ipsum reversus ab viral inferno, nam rick grimes malum cerebro. De
    carne lumbering animata corpora quaeritis. Summus brains sit​​, morbo vel
    maleficia? De apocalypsi gorger omero undead survivor dictum mauris. Hi
    mindless mortuis soulless creaturas, imo evil stalking monstra adventus resi
    dentevil vultus comedat cerebella viventium. Qui animated corpse, cricket
    bat max brucks terribilem incessu zomby. The voodoo sacerdos flesh eater,
    suscitat mortuos comedere carnem virus. Zonbi tattered for solum oculi eorum
    defunctis go lum cerebro.
  </p>
</div>
",
  "react": "<Box>
  <Paragraph color=\\"red\\">none</Paragraph>
  <Paragraph lineHeight=\\"none\\">
    Zombie ipsum reversus ab viral inferno, nam rick grimes malum cerebro. De
    carne lumbering animata corpora quaeritis. Summus brains sit​​, morbo vel
    maleficia? De apocalypsi gorger omero undead survivor dictum mauris. Hi
    mindless mortuis soulless creaturas, imo evil stalking monstra adventus resi
    dentevil vultus comedat cerebella viventium. Qui animated corpse, cricket
    bat max brucks terribilem incessu zomby. The voodoo sacerdos flesh eater,
    suscitat mortuos comedere carnem virus. Zonbi tattered for solum oculi eorum
    defunctis go lum cerebro.
  </Paragraph>
  <Paragraph color=\\"orange\\">small</Paragraph>
  <Paragraph lineHeight=\\"small\\">
    Zombie ipsum reversus ab viral inferno, nam rick grimes malum cerebro. De
    carne lumbering animata corpora quaeritis. Summus brains sit​​, morbo vel
    maleficia? De apocalypsi gorger omero undead survivor dictum mauris. Hi
    mindless mortuis soulless creaturas, imo evil stalking monstra adventus resi
    dentevil vultus comedat cerebella viventium. Qui animated corpse, cricket
    bat max brucks terribilem incessu zomby. The voodoo sacerdos flesh eater,
    suscitat mortuos comedere carnem virus. Zonbi tattered for solum oculi eorum
    defunctis go lum cerebro.
  </Paragraph>
  <Paragraph color=\\"yellow\\">normal</Paragraph>
  <Paragraph lineHeight=\\"normal\\">
    Zombie ipsum reversus ab viral inferno, nam rick grimes malum cerebro. De
    carne lumbering animata corpora quaeritis. Summus brains sit​​, morbo vel
    maleficia? De apocalypsi gorger omero undead survivor dictum mauris. Hi
    mindless mortuis soulless creaturas, imo evil stalking monstra adventus resi
    dentevil vultus comedat cerebella viventium. Qui animated corpse, cricket
    bat max brucks terribilem incessu zomby. The voodoo sacerdos flesh eater,
    suscitat mortuos comedere carnem virus. Zonbi tattered for solum oculi eorum
    defunctis go lum cerebro.
  </Paragraph>
  <Paragraph color=\\"pink\\">medium</Paragraph>
  <Paragraph lineHeight=\\"medium\\">
    Zombie ipsum reversus ab viral inferno, nam rick grimes malum cerebro. De
    carne lumbering animata corpora quaeritis. Summus brains sit​​, morbo vel
    maleficia? De apocalypsi gorger omero undead survivor dictum mauris. Hi
    mindless mortuis soulless creaturas, imo evil stalking monstra adventus resi
    dentevil vultus comedat cerebella viventium. Qui animated corpse, cricket
    bat max brucks terribilem incessu zomby. The voodoo sacerdos flesh eater,
    suscitat mortuos comedere carnem virus. Zonbi tattered for solum oculi eorum
    defunctis go lum cerebro.
  </Paragraph>
  <Paragraph color=\\"cyan\\">large</Paragraph>
  <Paragraph lineHeight=\\"large\\">
    Zombie ipsum reversus ab viral inferno, nam rick grimes malum cerebro. De
    carne lumbering animata corpora quaeritis. Summus brains sit​​, morbo vel
    maleficia? De apocalypsi gorger omero undead survivor dictum mauris. Hi
    mindless mortuis soulless creaturas, imo evil stalking monstra adventus resi
    dentevil vultus comedat cerebella viventium. Qui animated corpse, cricket
    bat max brucks terribilem incessu zomby. The voodoo sacerdos flesh eater,
    suscitat mortuos comedere carnem virus. Zonbi tattered for solum oculi eorum
    defunctis go lum cerebro.
  </Paragraph>
  <Paragraph color=\\"green\\">xlarge</Paragraph>
  <Paragraph lineHeight=\\"xlarge\\">
    Zombie ipsum reversus ab viral inferno, nam rick grimes malum cerebro. De
    carne lumbering animata corpora quaeritis. Summus brains sit​​, morbo vel
    maleficia? De apocalypsi gorger omero undead survivor dictum mauris. Hi
    mindless mortuis soulless creaturas, imo evil stalking monstra adventus resi
    dentevil vultus comedat cerebella viventium. Qui animated corpse, cricket
    bat max brucks terribilem incessu zomby. The voodoo sacerdos flesh eater,
    suscitat mortuos comedere carnem virus. Zonbi tattered for solum oculi eorum
    defunctis go lum cerebro.
  </Paragraph>
</Box>;
",
  "title": "Line Height",
}
`;

exports[`Site: Paragraph Sizes 1`] = `
Object {
  "docs": "Paragraphs can be customize to use several different sizes.",
  "html": "<div class=\\"drac-box\\">
  <p class=\\"drac-text drac-text-xs drac-line-height drac-text-white\\">xsmall</p>
  <p class=\\"drac-text drac-text-sm drac-line-height drac-text-white\\">small</p>
  <p class=\\"drac-text drac-line-height drac-text-white\\">medium</p>
  <p class=\\"drac-text drac-text-lg drac-line-height drac-text-white\\">large</p>
</div>
",
  "react": "<Box>
  <Paragraph size=\\"xsmall\\">xsmall</Paragraph>
  <Paragraph size=\\"small\\">small</Paragraph>
  <Paragraph size=\\"medium\\">medium</Paragraph>
  <Paragraph size=\\"large\\">large</Paragraph>
</Box>;
",
  "title": "Sizes",
}
`;

exports[`Site: Paragraph Weights 1`] = `
Object {
  "docs": "Paragraphs can be defined in different font weights.",
  "html": "<div class=\\"drac-box\\">
  <p class=\\"drac-text drac-line-height drac-text-white\\">normal</p>
  <p class=\\"drac-text drac-text-semibold drac-line-height drac-text-white\\">
    semibold
  </p>
  <p class=\\"drac-text drac-text-bold drac-line-height drac-text-white\\">bold</p>
</div>
",
  "react": "<Box>
  <Paragraph weight=\\"normal\\">normal</Paragraph>
  <Paragraph weight=\\"semibold\\">semibold</Paragraph>
  <Paragraph weight=\\"bold\\">bold</Paragraph>
</Box>;
",
  "title": "Weights",
}
`;
